dioptas/widgets/plot_widgets/NormalizedImageItem.py

```diff
@@ -0,0 +1,194 @@
+        """Returns the image data array
+
+        :param copy: False to return the array used internally: do not modify!
+        """
+        if self.__rawImage is None:
+            return None
+        return np.array(self.__rawImage, copy=copy)
+
+    def getNormalization(self) -> str:
+        """Returns the currently used normalization"""
+        return self.__normalization
+
+    def setNormalization(self, normalization: str):
+        """Set the data normalization to use"""
+        if normalization not in self._NORMALIZATIONS:
+            raise ValueError(f"Unsupported normalization: {normalization}")
+
+        if normalization == self.__normalization:
+            return
+
+        # Get levels **before** changing the normalization
+        levels = self.getLevels()
+        self.__normalization = normalization
+        self.setImage(image=self.__rawImage, levels=levels)
+
+    def _getNorm(self) -> Normalization:
+        return self._NORMALIZATIONS[self.getNormalization()]
+
+    def setImage(self, image=None, *args, **kwargs):
+        if image is None:
+            return super().setImage(None, *args, **kwargs)
+
+        self.__rawImage = image
+        normalizedImage = self._getNorm().apply(image)
+        return super().setImage(normalizedImage, *args, **kwargs)
+
+    def getLevels(self):
+        levels = super().getLevels()
+        return self._getNorm().revert(levels)
+
+    def setLevels(self, levels, update=True):
+        normalizedLevels = self._getNorm().apply(levels)
+        return super().setLevels(normalizedLevels, update)
+
+    @contextmanager
+    def _useAsImage(self, image: Optional[np.ndarray]):
+        """Context to temporarily use provided image as image attribute
+
+        Used when calling ImageItem methods that needs to access a different
+        image than the transformed one.
+        """
+        previousImage = self.image
+        self.image = image
+        try:
+            yield
+        finally:
+            self.image = previousImage
+
+    def quickMinMax(self, *args, **kwargs):
+        with self._useAsImage(self.__rawImage):
+            return super().quickMinMax(*args, **kwargs)
+
+    def getHistogram(self, *args, **kwargs):
+        if self.__rawImage is None:
+            return super().getHistogram(*args, **kwargs)
+
+        # Filter invalid values for current normalization
+        invalid = self._getNorm().invalid(self.__rawImage)
+        if not np.any(invalid):
+            filteredImage = self.__rawImage
+
+        else:  # Replace invalid values with NaNs
+            dtype = self.__rawImage.dtype
+            if not np.issubdtype(dtype, np.floating):
+                dtype = np.float64
+            filteredImage = np.array(self.__rawImage, copy=True, dtype=dtype)
+
+            filteredImage[invalid] = np.nan
+
+        with self._useAsImage(filteredImage):
+            return super().getHistogram(*args, **kwargs)
```

Comment:
This computes the histogram on valid data for the current normalization.

Reply:
This maybe be not needed not that histogram is computed from image data since PR #146